x-pack/plugins/lens/public/app_plugin/csv_download_provider/csv_download_provider.tsx

```diff
@@ -30,21 +40,29 @@ async function downloadCSVs({
   title,
   formatFactory,
   uiSettings,
-  columnsSorting,
+  tablesToShare = [],
+  sortedColumns,
+  columnSorting,
 }: {
   title: string;
   activeData: TableInspectorAdapter;
   formatFactory: FormatFactory;
   uiSettings: IUiSettingsClient;
-  columnsSorting?: string[];
+  tablesToShare?: string[];
+  sortedColumns?: string[];
+  columnSorting?: EuiDataGridColumnSortingConfig[];
 }) {
   if (!activeData) {
     if (window.ELASTIC_LENS_CSV_DOWNLOAD_DEBUG) {
       window.ELASTIC_LENS_CSV_CONTENT = undefined;
     }
     return;
   }
-  const datatables = Object.values(activeData);
+  const seletedDatatables = Object.entries(activeData)
+    .filter(([id]) => tablesToShare.includes(id))
+    .map(([, table]) => table);
+  const datatables = seletedDatatables.length > 0 ? seletedDatatables : Object.values(activeData);
```

Comment:
Allow selecting a subset of tables to export. Ensures at least 1 table is available to export otherwise exports all available tables.

Reply:
This is now handled in `vis. getExportDatatables`, not need to tell filter tables, now we just return the tables to be exported.